### PR TITLE
 refactor(core/webidl): upgrade to webidl2.js 16 

### PR DESCRIPTION
**Pull request by saschanaz**
👍

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1418,7 +1418,7 @@
     },
     "array-equal": {
       "version": "1.0.0",
-      "resolved": "http://registry.npmjs.org/array-equal/-/array-equal-1.0.0.tgz",
+      "resolved": "https://registry.npmjs.org/array-equal/-/array-equal-1.0.0.tgz",
```
Comment (saschanaz) [inline package-lock.json]: back and forth 🤔

Comment (saschanaz) [reply package-lock.json]: 6.4.1, right? So this is a mystery...

**Comment by saschanaz** [comment]
Thank you for your quick review 👍

File: package-lock.json (hunk 1 above)
Comment (marcoscaceres) [reply package-lock.json]: 6.4.1... correct. Super odd.

Comment (saschanaz) [reply package-lock.json]: Looks like this is a known issue:

https://npm.community/t/npm-install-downgrading-resolved-packages-from-https-to-http-registry-in-package-lock-json/1818
https://npm.community/t/some-packages-have-dist-tarball-as-http-and-not-https/285

The question is why I get HTTPS then? 🤔🤔🤔🤔🤔

**Pull request merged**